parameter/criterion/src/Criterion.cpp

```diff
@@ -0,0 +1,255 @@
+}
+
+std::string Criterion::getFormattedDescription(bool bWithTypeInfo, bool bHumanReadable) const
+{
+    std::string strFormattedDescription;
+    if (bHumanReadable) {
+
+        if (bWithTypeInfo) {
+
+            // Display type info
+            CUtility::appendTitle(strFormattedDescription, mName + ":");
+
+            // States
+            strFormattedDescription += "Possible states ";
+
+            // Type Kind
+            strFormattedDescription += "(";
+            strFormattedDescription += getKind();
+            strFormattedDescription += "): ";
+
+            // States
+            strFormattedDescription += listPossibleValues() + "\n";
+
+            // Current State
+            strFormattedDescription += "Current state";
+        } else {
+            // Name only
+            strFormattedDescription = mName;
+        }
+
+        // Current State
+        strFormattedDescription += " = " + getFormattedState();
+    } else {
+        // Name
+        strFormattedDescription = "Criterion name: " + mName;
+
+        if (bWithTypeInfo) {
+            // Type Kind
+            strFormattedDescription += ", type kind: ";
+            strFormattedDescription +=  getKind();
+        }
+
+        // Current State
+        strFormattedDescription += ", current state: " + getFormattedState();
+
+        if (bWithTypeInfo) {
+            // States
+            strFormattedDescription += ", states: " + listPossibleValues();
+        }
+    }
+    return strFormattedDescription;
+}
+
+void Criterion::toXml(CXmlElement& xmlElement, CXmlSerializingContext& serializingContext) const
+{
+    (void)serializingContext;
+    xmlElement.setAttributeString("Value", getFormattedState());
+    xmlElement.setAttributeString("Name", mName);
+    xmlElement.setAttributeString("Kind", getKind());
+
+    for (auto& value : mValues) {
+        CXmlElement childValueElement;
+
+        xmlElement.createChild(childValueElement, "Value");
+        childValueElement.setTextContent(value);
+    }
+}
+
+const std::string Criterion::getKind() const
+{
+    return "Exclusive";
+}
+
+std::string Criterion::getFormattedState() const
+{
+    assert(!mState.empty());
+    return *mState.begin();
+}
+
+std::string Criterion::listPossibleValues() const
+{
+    std::string possibleValues = "{";
+
+    // Get comma separated list of values
+    bool first = true;
+    for (auto& value : mValues) {
+
+        if (first) {
+            first = false;
+        } else {
+            possibleValues += ", ";
+        }
+        possibleValues += value;
+    }
+    possibleValues += "}";
+
+    return possibleValues;
+}
+
+bool Criterion::isValueAvailable(const Value &value) const
+{
+    return mValues.count(value) == 1;
+}
+
+bool Criterion::match(const std::string& method, const State& state) const
+{
+    return mMatchMethods.at(method)(state);
+}
+
+bool Criterion::isMatchMethodAvailable(const std::string& method) const
+{
+    return mMatchMethods.count(method) == 1;
```

Comment:
suboptimal/strange semantic. proposal:

    return mMatchMethods.find(method) != mMatchMethods.end()

Same stuff for "isValueAvailable".